File: HSG_HAI_DUONG_22_23/contestants/huynh_minh_thuan/BAI2.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ifstream inp("BAI2.inp");
    ofstream out("BAI2.out");

    int n, i = 0;
    long long c = 0, ans = 0;
    string res;

    inp >> n >> res;

    while (i < n) {
        if (res[i] == '0') i++;
        else while (res[i] == '1') c++, i++;

        ans += c * (c + 1) / 2;
        c = 0;
    }

    out << ans;

    inp.close();
    out.close();

    return 0;
}

File: HSG_HAI_DUONG_22_23/contestants/nguyen_the_khoi/bai2.py
with open ("bai2.inp","r") as f:
    n=int(f.readline())
    a=f.readline()
    f.close()
s=0
count=0
for i in a:
    if i=='1':
        count+=1
        s+=count
    else:
        count=0
with open ("bai2.out","w") as f:
    f.write(str(s))
    f.close()